src/codeu/chat/client/core/MessageContext.java

```diff
@@ -18,18 +18,22 @@
 import java.util.Iterator;
 
 import codeu.chat.common.BasicView;
+import codeu.chat.common.ConversationHeader;
 import codeu.chat.common.Message;
-import codeu.chat.util.Serializer;
-import codeu.chat.util.Serializers;
+import codeu.chat.common.User;
 import codeu.chat.util.Uuid;
 
 public final class MessageContext {
   
   public final Message message;
-  public final BasicView view;
+  private final ConversationHeader conversation;
+  private final User user;
+  private final BasicView view;
 
-  public MessageContext(Message message, BasicView view) {
+  public MessageContext(Message message, ConversationHeader conversation, User user, BasicView view) {
```

Comment:
I'll say it once and I'll say it again: any variable you need to get conversation and user is in the Message class itself.